hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -797,6 +931,24 @@ def resolve_Callable(thing):
     # use of keyword arguments and we'd rather not force positional-only.
     if not thing.__args__:  # pragma: no cover  # varies by minor version
         return st.functions()
+
+    # Concatenate and ParamSpec can never be registered or resolved at Callable arguments
+    # if it is a single argument it will not be iterable
+    if not isinstance(thing.__args__[0], Iterable):
+        thing_args = thing.__args__[:-1]  # for python 3.8
+    else:
+        thing_args = thing.__args__[0]
+    for stuff in thing_args:
+        if (
+            getattr(stuff, "__origin__", None) in ConcatenateTypes + ParamSpecTypes
+        ) or (type(stuff) in ConcatenateTypes + ParamSpecTypes):
+            raise InvalidArgument(f"{stuff} cannot be arguments in Callables.")
+    # TypeGuard can never be returned by Callable types
+    if getattr(thing.__args__[-1], "__origin__", None) in TypeGuardTypes:
+        raise InvalidArgument(
+            f"Return type of Callables cannot be {thing.__args__[-1]}."
+        )
```

Comment:
Similarly, "Hypothesis cannot yet construct a strategy for callables which are PEP-647 TypeGuards(got {thing.__args__[-1]!r}).  Consider using an explicit strategy, or opening an issue."